src/game/Maps/Map.cpp

```diff
@@ -909,16 +909,18 @@ void Map::Remove(T* obj, bool remove)
 
     m_objRemoveList.insert(obj->GetObjectGuid());
 
-    obj->ResetMap();
     if (remove)
     {
         // if option set then object already saved at this moment
         if (!sWorld.getConfig(CONFIG_BOOL_SAVE_RESPAWN_TIME_IMMEDIATELY))
             obj->SaveRespawnTime();
 
+        obj->ResetMap();
```

Comment:
I think this is unnecessary because `obj` is getting deleted right afterwards anyway, but maybe there was method to the madness, hence I kept it in both branches.

Reply:
I vote for removing this, since the destructor called by `delete` *can* invoke `m_currMap->RemoveFromOnEventNotified(this);`